### ISSUE TITLE
TestParse fails on Mac

### ISSUE THREAD

**Issue by janiversen** [subsurface/subsurface#1485]
### Describe the issue:

- [ x] Bug
- [ ] Change request
- [ ] New feature request
- [ ] Discussion request

### Issue long description:

When running TestParse on macOS 10.13 it fails with:

Strange depth reading NaN m
Strange depth reading NaN m
Strange depth reading NaN m
Strange depth reading NaN m
export_dives_xslt with stylesheet uddf-export.xslt
FAIL!  : TestParse::testExport() Compared values are not the same
   Actual   (written.takeFirst().trimmed()): "<depth>2,34</depth>"
   Expected (readin.takeFirst().trimmed()) : "<depth>0</depth>"
   Loc: [/Users/jani/develop/src/subsurface/tests/testparse.cpp(446)]
PASS   : TestParse::parseDL7()
PASS   : TestParse::cleanupTestCase()
Totals: 12 passed, 2 failed, 0 skipped, 0 blacklisted, 3967ms
********* Finished testing of TestParse *********

Seems like a problem.

### Steps to reproduce:

1) run "make check" in subsurface
2) cd build/tests
3) ./TestParse

**Comment by dirkhh** [comment]
I assume that's with latest master?
Here on macOS 10.11 all tests parse. Oddly, sometimes `TestGpsCoords` crashes when run from `make check`, but when run individually it passes as well. So I wonder what's different about your setup...

**Comment by mturkia** [comment]
I'll need at least the testuddfexport.uddf and testuddfexport2.uddf files produced by this test. This might give a hint of what is wrong. However, it does sound weird if this is working on one version of macOS but not on another one. (And I don't have access to any mac so cannot test myself.)

**Comment by janiversen** [comment]
my Mac is 10.13.5 and I used a master about 2 days old.  I just rerun "./TraceParse" in build/tests, with a master I updated this morning.

You will find the screen output in TraceParse.log and (I hope all the other generated files).

Remark had to give the file a .txt ending to be accepted, remove that and do a normal tar -xzf.

[tests.tgz.txt](https://github.com/Subsurface-divelog/subsurface/files/2181227/tests.tgz.txt)
